src/physics/utilities/solver_config.hpp

```diff
@@ -64,14 +68,24 @@ enum class NonlinearSolver
 };
 
 /**
- * @brief Preconditioning method
+ * @brief Stores the information required to configure a HypreSmoother
  */
-enum class Preconditioner
-{
-  Jacobi,
-  BoomerAMG
+struct HypreSmootherPrec {
+  mfem::HypreSmoother::Type type;
+};
+
+/**
+ * @brief Stores the information required to configure a HypreBoomerAMG preconditioner
+ */
+struct HypreBoomerAMGPrec {
+  mfem::ParFiniteElementSpace* pfes = nullptr;
 };
 
+/**
+ * @brief Preconditioning method
+ */
+using Preconditioner = std::variant<HypreSmootherPrec, HypreBoomerAMGPrec>;
```

Comment:
Is it considered better to use variants instead of base class pointers now?

Reply:
I found [this discussion](https://stackoverflow.com/questions/59784261/stdvariant-vs-pointer-to-base-class-for-heterogeneous-containers-in-c) on stack overflow. One thing to keep in mind is variant could grow quite a bit as we add more non-hypre preconditioners.

Reply:
In my opinion it depends on the situation - I think interfaces can be useful when there is lots of commonality of behavior, but in this case I would expect variant members to just be small structs/dataclasses with little commonality of state or behavior.

The lack of commonality could also require a fair amount of dynamic_casting if this were implemented with inheritance as the exact preconditioner type must be known for the AMG prec to be set up correctly.